### DISCUSSION TITLE
Ability to disable new signups, or disable login

### DISCUSSION THREAD

**Discussion by jivanyatra** [enchant97/note-mark#116] [Ideas]
I self host a lot of my stuff on different servers. I either have:

- basic auth setup alongside SSL, and I turn off the app's login mechanism
- SSL setup with nginx proxy, and use the app's login mechanism, but I disable the signup function

I think it'd be useful to do these things. Not sure if the capability is already there, but I didn't see anything in the docs. Figured I'd discuss.

**Comment by enchant97** [comment]
Thanks for your suggestions, currently Note Mark has the ability to disable new signups through the `ALLOW_SIGNUP` by setting it to `false`. I'm not sure how to implement the ability to disable the in-built login to use external, however will look into it at some point.

**Reply by jivanyatra** [reply]
> through the `ALLOW_SIGNUP` by setting it to `false`
Good to know! ~~Is that an environment variable? I'm happy to update the docs to include this.~~ Saw it in the docs, perfect!

>however will look into it at some point
Perhaps disabling the in-built login could work the same way, via an environment variable. I understand you're not prioritizing it, that's certainly fine by me.

Thank you!

**Reply by vnghia** [reply]
Hello, If I set the `ALLOW_SIGNUP` to `false`, is there a way to create an admin account without signup or I have to create an admin account first and disable signup later ?

**Reply by enchant97** [reply]
@vnghia there is no admin account functionality, all users created via signup screen have equal rights.

Ensure all users have signed-up before setting `ALLOW_SIGNUP=false`. In future there will be a cli command added to allow for accounts to be created bypassing this.

**Reply by vnghia** [reply]
I see.

> In future there will be a cli command added to allow for accounts to be created bypassing this.

Great! I think this will be beneficial for users with declarative infrastructure, like if you use `docker-compose` with `ALLOW_SIGNUP=false`, you will have to deploy twice, one with `ALLOW_SIGNUP=true` to let the users sign up and the second deployment with `ALLOW_SIGNUP=false`.

Thank you very much for this great software 😄
